src/main.rs

```diff
@@ -142,8 +162,7 @@ fn run(
     // This object contains all of the state about what's being displayed. It's
     // wrapped in a clonable mutex since both the I/O loop and display need to
     // access it.
-    let mut terminal = Term::new(&config, display.size().to_owned());
-    terminal.set_logger_proxy(logger_proxy.clone());
+    let terminal = Term::new(&config, display.size().to_owned(), message_bar);
```

Comment:
I'll be honest, I was kinda hoping that the terminal would create it's own message bar. I think that's because of the name `MessageBar`, instead it's really more of a message buffer, that also knows how to format lines for a bar. Not a really bad thing, but a tiny bit odd.

Reply:
I honestly don't think `MessageBar` is a great name. I feel like it might be the right choice to just rename that. Maybe `MessageBuffer` is more appropriate? Though that wouldn't really be obvious as to what it's buffering messages for.

Maybe `StatusLineBuffer`?

Reply:
Decided to go for `MessageBuffer`.